### ISSUE TITLE
does not sort imports in functions

### ISSUE THREAD

**Issue by squfrans** [timothycrosley/isort#302]
minimal example:

```py
def foo():
    from uuid import uuid4
    from datetime import datetime, timedelta
    from base64 import urlsafe_b64encode
```

**Comment by timothycrosley** [comment]
Hi @squfrans,

Thanks for reporting and documenting this issue! Unfortunately, I don't think I can support this at this time. This issue is, that when a user adds imports into a function, there really *already* breaking most standards of coding, and doing something outside of the normal design. This makes it unclear how to resolve there issue - do we move the imports to the top of the file and make them conform to most coding standards? We could, but most users only put imports into functions as a last resort, when they run into a circular dependency, so then do we just assume that the correct course of action is to keep them in the import but sort them? Again we could, but the fact there in a function, raises the possibility that there is magic happening (there adding to system.path, depending on the order of imports to avoid dependency issues), this means that sorting could then break their magic. That said, I am 100% open to an implementation of this sent in as a pull-request if it is proved to be relatively safe, and optional.

Thanks!

~Timothy

**Comment by squfrans** [comment]
good points.  indeed we have these because of circular imports.

however i use Isort mostly in vim visual blocks, not entire files.
as such, i know exactly when i want to sort and what.

so a naive implementation could simply take a setting like `inside_functions`,
default to `false` and let the big boys shoot themselves in the foot :]

**Comment by blueyed** [comment]
I think that isort should not move the imports out of the function, but just sort them inside.
When `isort` is used on the whole file those could still be ignored by default.

My use case (from #398) is the same as @squfrans' - sorting a visual block in Vim.
My current workaround is to first remove the indent, call the isort command and re-apply the indenting.

**Comment by xi** [comment]
This seems to be fixed in 5.0

**Comment by jgod** [comment]
Does anyone know if it's possible to turn off sorting, or even just the sections, of imports inside functions?
